### ISSUE TITLE
fhir_resource_history sorts in the wrong order

### ISSUE THREAD

**Issue by vadi2** [fhirbase/fhirbase-plv8#107]
https://hl7-fhir.github.io/http.html#history mentions that the sort order is newest to last ("sorted with oldest versions last"), but in fhirbase 1.3.0.12 the older version is getting printed first. 

The first resource's lastUpdated is 2016-04-26T19:19:**33**.596Z and the second ones is 2016-04-26T19:19:**35**.447Z in this fhirbase response:

``` json
{
  "meta": {
    "lastUpdated": "2016-04-26T19:29:03.868Z"
  },
  "total": 3,
  "type": "history",
  "resourceType": "Bundle",
  "entry": [
    {
      "request": {
        "url": "AllergyIntolerance",
        "method": "POST"
      },
      "resource": {
        "substance": {
          "coding": [
            {
              "code": "314422",
              "display": "ALLERGENIC EXTRACT, PENICILLIN",
              "system": "http://www.nlm.nih.gov/research/umls/rxnorm"
            }
          ]
        },
        "recorder": {
          "reference": "Practitioner/exampleLQSyEt"
        },
        "status": "unconfirmed",
        "resourceType": "AllergyIntolerance",
        "text": {
          "status": "generated",
          "div": "<div>Mar 1, 2010: ALLERGENIC EXTRACT, PENICILLIN</div>"
        },
        "patient": {
          "reference": "Patient/exampleOqPgE"
        },
        "meta": {
          "versionId": "42ba84e7-067d-4808-9dc9-c4fffd3cfe30",
          "extension": [
            {
              "url": "fhir-request-method",
              "valueString": "POST"
            },
            {
              "url": "fhir-request-uri",
              "valueUri": "AllergyIntolerance"
            }
          ],
          "lastUpdated": "2016-04-26T19:19:33.596Z"
        },
        "identifier": [
          {
            "value": "II35318786125",
            "system": "http://happyvalley.com/allergyintolerance"
          }
        ],
        "category": "medication",
        "id": "II35318786125",
        "criticality": "CRITH",
        "reaction": [
          {
            "manifestation": [
              {
                "coding": [
                  {
                    "code": "247472004",
                    "display": "Hives",
                    "system": "http://snomed.info/sct"
                  }
                ]
              }
            ]
          }
        ],
        "recordedDate": "2010-03-01"
      }
    },
    {
      "request": {
        "url": "AllergyIntolerance",
        "method": "PUT"
      },
      "resource": {
        "substance": {
          "coding": [
            {
              "code": "314422",
              "display": "ALLERGENIC EXTRACT, PENICILLIN",
              "system": "http://www.nlm.nih.gov/research/umls/rxnorm"
            }
          ]
        },
        "recorder": {
          "reference": "Practitioner/exampleLQSyEt"
        },
        "status": "confirmed",
        "resourceType": "AllergyIntolerance",
        "text": {
          "status": "generated",
          "div": "<div>Mar 1, 2010: ALLERGENIC EXTRACT, PENICILLIN</div>"
        },
        "patient": {
          "reference": "Patient/exampleOqPgE"
        },
        "meta": {
          "versionId": "027d032c-a960-45cc-b6e2-f5cce8c49c82",
          "extension": [
            {
              "url": "fhir-request-method",
              "valueString": "PUT"
            },
            {
              "url": "fhir-request-uri",
              "valueUri": "AllergyIntolerance"
            }
          ],
          "lastUpdated": "2016-04-26T19:19:35.447Z"
        },
        "identifier": [
          {
            "value": "II35318786125",
            "system": "http://happyvalley.com/allergyintolerance"
          }
        ],
        "category": "medication",
        "id": "II35318786125",
        "criticality": "CRITU",
        "reaction": [
          {
            "manifestation": [
              {
                "coding": [
                  {
                    "code": "247472004",
                    "display": "Hives",
                    "system": "http://snomed.info/sct"
                  }
                ]
              }
            ]
          }
        ],
        "recordedDate": "2010-03-01"
      }
    },
    {
      "request": {
        "url": "AllergyIntolerance",
        "method": "DELETE"
      }
    }
  ]
}
```

Test case - create this resource first:

``` json
{
  "resourceType" : "AllergyIntolerance",
  "id" : "II35318786125",
  "text" : {
    "status" : "generated",
    "div" : "<div>Mar 1, 2010: ALLERGENIC EXTRACT, PENICILLIN</div>"
  },
  "identifier" : [
    {
      "system" : "http://happyvalley.com/allergyintolerance",
      "value" : "II35318786125"
    }
  ],
  "recordedDate" : "2010-03-01",
  "recorder" : {
    "reference" : "Practitioner/exampleLQSyEt"
  },
  "patient" : {
    "reference" : "Patient/exampleOqPgE"
  },
  "substance" : {
    "coding" : [
      {
        "system" : "http://www.nlm.nih.gov/research/umls/rxnorm",
        "code" : "314422",
        "display" : "ALLERGENIC EXTRACT, PENICILLIN"
      }
    ]
  },
  "status" : "unconfirmed",
  "criticality" : "CRITH",
  "category" : "medication",
  "reaction" : [
    {
      "manifestation" : [
        {
          "coding" : [
            {
              "system" : "http://snomed.info/sct",
              "code" : "247472004",
              "display" : "Hives"
            }
          ]
        }
      ]
    }
  ]
}
```

Then update it with: 

``` json
{
  "resourceType" : "AllergyIntolerance",
  "id" : "II35318786125",
  "text" : {
    "status" : "generated",
    "div" : "<div>Mar 1, 2010: ALLERGENIC EXTRACT, PENICILLIN</div>"
  },
  "identifier" : [
    {
      "system" : "http://happyvalley.com/allergyintolerance",
      "value" : "II35318786125"
    }
  ],
  "recordedDate" : "2010-03-01",
  "recorder" : {
    "reference" : "Practitioner/exampleLQSyEt"
  },
  "patient" : {
    "reference" : "Patient/exampleOqPgE"
  },
  "substance" : {
    "coding" : [
      {
        "system" : "http://www.nlm.nih.gov/research/umls/rxnorm",
        "code" : "314422",
        "display" : "ALLERGENIC EXTRACT, PENICILLIN"
      }
    ]
  },
  "status" : "confirmed",
  "criticality" : "CRITU",
  "category" : "medication",
  "reaction" : [
    {
      "manifestation" : [
        {
          "coding" : [
            {
              "system" : "http://snomed.info/sct",
              "code" : "247472004",
              "display" : "Hives"
            }
          ]
        }
      ]
    }
  ]
}
```

The difference is in the status: confirmed and status: unconfirmed values.

Credit for discovering this goes to Touchstone.

**Comment by danil** [comment]
@vadi2  fixed in v1.3.0.13
https://github.com/fhirbase/fhirbase-plv8/blob/v1.3.0.13/test/fhir/history_spec.coffee#L39
if not, feel free to reopen

**Comment by vadi2** [comment]
Thanks, works!
